components/twitter/twitter.app.js

```diff
@@ -818,19 +885,212 @@ module.exports = {
         console.log(err) // TODO
       }
     },
-    async search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode) {   
-      const query = querystring.stringify({
-        q,
-        since_id,
-        tweet_mode,
-        count,
+    async getFollowers(screen_name) {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/followers/ids.json?`,
+        params: {
+          screen_name,
+          stringify_ids: true,
+        }
+      })).data.ids
+    },
+    async getLikedTweets(opts = {}) {
+      const {
+        screen_name,
+        count = 200,
+        tweet_mode = 'extended',
+      } = opts
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/favorites/list.json`,
+        params: {
+          screen_name,
+          count,
+          tweet_mode,
+        }
+      })).data
+    },
+    async lookupUsers(userIdArray) {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/users/lookup.json`,
+        params: {
+          user_id: userIdArray.join(),
+        }
+      })).data
+    },
+    async search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode, max_id) {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/search/tweets.json`,
+        params: {
+          q,
+          since_id,
+          max_id,
+          tweet_mode,
+          count,
+          result_type,
+          lang,
+          locale,
+          geocode,
+        }
+      }))
+    },
+    async getTrendLocations() {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/trends/available.json`,
+      })).data
+    },
+    async getTrends(opts = {}) {   
+      const {
+        id = 1,
+      } = opts
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/trends/place.json`,
+        params: {
+          id,
+        }
+      })).data
+    },
+    async searchHelper(opts = {}) {
+      const tweets = []
+
+      const {
+        tweet_mode = 'extended',
         result_type,
+        count = 100,
         lang,
         locale,
         geocode,
-      })
+        enrichTweets = true,
+        includeReplies = true,
+        includeRetweets = true,
+      } = opts
+
+      let { q, max_id, since_id = 0 } = opts
+      let min_id
+
+      if(includeReplies === false) {
+        q = `${q} -filter:replies`
+      }
+  
+      if(includeRetweets === false) {
+        q = `${q} -filter:nativeretweets`
+      }
+  
+      const response = await this.search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode, max_id)
+      
+      //console.log(response)
+      if (_.get(response, 'status', 'Error') === 200) {
+        if (!max_id) {
+          max_id = since_id || 0
+        }
+    
+        for (let tweet of response.data.statuses) {
+          if (tweet.id_str !== since_id && tweet.id_str !== max_id) {
```

Comment:
why are you comparing string and numbers here? (or why are you setting max_id / since_id to `0` by default)

Reply:
I updated this